src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -148,13 +129,20 @@ public interface IChainedHeaderTree
         /// <param name="block">The block.</param>
         /// <returns>Newly created and connected chained header for the specified block.</returns>
         ChainedHeader CreateChainedHeaderWithBlock(Block block);
+
+        /// <summary>
+        /// Get the block and its chained header if it exists.
+        /// If the header is not in the tree returns <c>null</c>, the <see cref="ChainedHeaderBlock.Block"/> may also be null.
```

Comment:
This rather belongs to returns not to the summary

Also use see for the second null

Finally, I would at the information that the block is null when the block data has not yet been downloaded or if the block data has been persisted to the database and removed from the memory.

Reply:
I can add the null to the return as well but normally the summery is the first thing you read (and often the only thing users read)

Reply:
Please use the sections properly.